tools/clang/include/clang/Basic/DiagnosticGroups.td

```diff
@@ -783,6 +783,7 @@ def HLSLEffectsSyntax : DiagGroup<"effects-syntax">;
 def HLSLForRedefinition : DiagGroup<"for-redefinition">;
 def HLSLAttributeStatement : DiagGroup<"attribute-statement">;
 def HLSLAttributeType : DiagGroup<"attribute-type">;
+def HLSLEntryAttributeWithoutShaderAttrType : DiagGroup<"entry-attribute-type-without-shader-attribute-type">;
```

Comment:
It's kinda long. Maybe something as simple as "non-entry-attributes", "misplaced-attributes"?